### ISSUE TITLE
[Question] Bundle/split external dependencies only

### ISSUE THREAD

**Issue by richarddd** [evanw/esbuild#1781]
Hi,

Is there any way to create "two" bundles. One with my sources and another one with my dependencies?
The main bundle should import the dependencies bundle.

Regards
Richard

**Comment by hyrious** [comment]
https://github.com/hyrious/esbuild-split-vendors-example

**Comment by richarddd** [comment]
@hyrious i opened this issue again because it's not an optimal solution. This will bundle ALL external dependencies and not just the ones you are using create a vendor.js bundle that is very large (in my case 15x larger than when everything is bundled in one file)

**Comment by hyrious** [comment]
Yes this is just a quick solution but not optimal one. To analyze dependencies that are really in use, you must bundle once and look in the metafile, instead of just reading package.json.

I'll suggest using [vite](https://vitejs.dev) to do this work for you, since it hooks into build process and automatically split dependencies into a vendor.js.
